cirq-core/cirq/ops/matrix_gates_test.py

```diff
@@ -218,6 +218,66 @@ def test_two_qubit_diagram():
     )
 
 
+def test_named_single_qubit_diagram():
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+    m = np.array([[1, 1j], [1j, 1]]) * np.sqrt(0.5)
+    c = cirq.Circuit(cirq.MatrixGate(m, name='Foo').on(a), cirq.CZ(a, b))
+
+    assert re.match(
+        r"""
+a: ───Foo───@───
+            │
+b: ─────────@───
+    """.strip(),
+        c.to_text_diagram().strip(),
+    )
+
+    assert re.match(
+        r"""
+a   b
+│   │
+Foo │
+│   │
+@───@
+│   │
+    """.strip(),
+        c.to_text_diagram(transpose=True).strip(),
+    )
+
+
+def test_named_two_qubit_diagram():
+    a = cirq.NamedQubit('a')
+    b = cirq.NamedQubit('b')
+    c = cirq.NamedQubit('c')
+    c = cirq.Circuit(
+        cirq.MatrixGate(cirq.unitary(cirq.CZ), name='Foo').on(a, b),
+        cirq.MatrixGate(cirq.unitary(cirq.CZ), name='Bar').on(c, a),
+    )
+    assert re.match(
+        r"""
+a: ───Foo1───Bar2───
+      │      │
+b: ───Foo2───┼──────
+             │
+c: ──────────Bar1───
+    """.strip(),
```

Comment:
I _think_ I like the numbering though, I'm not sure about it. I guess the qubit order does matter, so it can be informative. In typical diagrams this info can be lost as we just draw a "box" around the participating qubits. However gates like CX have clear directionality in their notations. So let's keep it. 

I would separate the number from the name somehow...
`Foo[1]` 

Also, maybe use 0 indexing?

```
a: ───Foo[0]───Bar[1]───
      │        │
b: ───Foo[1]───┼──────
               │
c: ────────────Bar[0]───
```

Reply:
Will do.  I personally like the 0 based indexing more as well, but I just went with 1 based indexing because that is what happens in matrix form.

Reply:
@balopat finished implementing your suggestions.  What do you think?

Reply:
Oh I see what you mean, I guess, let's keep it with 1 based indexing to keep it consistent. 
Can you also run `check/format-incremental --apply`? I think we're good after that! Thanks!

Reply:
@balopat Done